[7주차] 곽성재 과제 제출합니다

kwakseongjae opened the pull request:
기나긴 버그와의 싸움.. 일주일만에 돌아는 가게 만들었습니다 ,, ㅠ
이번 프로젝트의 메인 Task는 
1. Recoil로 상태관리
2. 데이터를 js파일이 아닌 json파일로 처리 
3. 로컬스토리지를 이용하여 데이터 저장
4. 사진 첨부 기능 추가 
5. 잡다한 CSS 수정
입니다 ! 
생각보다 오래 붙들고 코드를 작성해서 딱 마감기한까지는 1~2번 항목만 구현했습니다. 
스터디 모임 전까지 3, 5번 항목 구현 및 리팩토링(주석 제거..)를 메인으로 진행하고, 
가능하다면 사진 첨부 기능까지 구현해 오겠습니다 :) 

Vercel 배포 링크입니다 !
https://7-react-routing-state-pi.vercel.app/

kwakseongjae committed:
FEAT: states 폴더 추가

kwakseongjae committed:
FEAT: UserProfile에 Recoil 적용

kwakseongjae committed:
FEAT: 친구 목록에 Recoil 적용 완료

kwakseongjae committed:
FEAT: ChattingList에 Recoil 적용

kwakseongjae committed:
FEAT: Chatting에 Recoil적용(일단 돌아는가는..)

kwakseongjae committed:
FIX: MessageBox에서 chatList가 온전히 매핑되지 않던 오류 수정

kwakseongjae committed:
FIX: 채팅시 발생하는 오류 임시 해결

kwakseongjae committed:
REFACTOR: chattingsData.json 파일 내용 수정
- 데이터 양이 부족하여 일부 추가
- 데이터 내용이 올바르지 않아 내용 수정

kwakseongjae committed:
REFACTOR: 불필요한 주석 및 파일 제거
- states/datas 폴더의 User.js, BasicChat.js 파일 제거
- 기존 App.js에서 사용하던 불필요한 경로 제거 -> 현재 components/Router.js가 해당 역할 수행
- components폴더의 FileInput.js와 ImgUploader.js 는 현재 사용하지 않지만 향후 추가할 기능이기 때문에 잔류

kwakseongjae committed:
FIX: 더보기 오류 해결
- 기존 states/datas/User.js 파일 삭제로 인해 발생한 오류 수정

kwakseongjae committed:
FEAT: utilities/localStorage.js 추가
- Recoil Effects를 이용해 로컬스토리지에 데이터 입력, 받아오기 기능 추가 시도
- MessageChattingView.js 에서 작성한 방식때문에 해당 기능 사용 불가 판단
- 다음 리팩토링을 통해 기능 구현 예정

ihatemondayy commented on src/styles/Chatting.styled.js:
```diff
@@ -0,0 +1,129 @@
+  flex-direction: column;
+`;
+
+export const ChattingImage = styled.img`
+  width: 50px;
+  height: 50px;
+  border-radius: 70%;
+  overflow: hidden;
+  margin: 15px 15px 0px 15px;
+`;
```
지금 flex 아이템보다 container의 크기가 작아서 아이템 크기가 줄어들어 프로필 img가 50*50px 유지를 못 하는 현상이 발생하는 것 같은데,  여기에 `flex-shrink: 0;` 속성 추가하시면 해결될 거예요

kwakseongjae committed:
FIX: 채팅창에서 프로필 이미지가 찌그러지는 버그 수정
- Chatting.styled.js파일 ChattingImage에 flex-shrink: 0; 추가

kwakseongjae committed:
FIX: 검색기능 사용 시 발생하는 오류 해결
- 친구, 채팅 페이지에서 검색 기능을 사용할 때 전달하는 함수 이름을 통일하지 못해서 발생한 오류
- searChat, searUser에서 searchItem으로 이름을 통일한 후 로직 수정

kwakseongjae committed:
STYLE: ProfileModal 텍스트 중앙 정렬
- h1태그와 p태그에 직접 style적용(text-align: center;)

kwakseongjae committed:
FEAT: Recoil effects를 이용한 로컬스토리지 구현
- 채팅 기록을 로컬스토리지에 저장하여 새로고침을 하거나 채팅리스트 페이지로 돌아가도 채팅 기록이 유지되도록 코드 작성

kimnabong2 commented on src/states/atoms/chattings.js:
여기 찾기 기능으로 fiter 깔끔하게 쓰셔서 괜시리 만족스럽네요 ㅋㅋㅋㅋㅋㅋㅋ
저도 이것저것 불러오는 부분을 처리하고 깔끔하게 바꿔야겠어요! 배우고 갑니다!

kimnabong2 commented on src/components/Chatting/MessageChattingView.js:
스크롤 기능 추가해도 좋을거 같아요!
도움 받았던 사이트 공유하고 갑니당!
https://minhyeong-jang.github.io/2020/01/17/js-scroll-bottom
나중에 타입스크립트로 변경하게 되면 도움될거 같아요

kimnabong2 commented on src/states/atoms/user.js:
```diff
@@ -0,0 +1,37 @@
+import userData from "../datas/userData.json";
+import { atom, selectorFamily } from "recoil";
+import friendsData from "../datas/friendsData.json";
+
+export const userState = atom({
+  key: "user",
+  default: userData.users[0],
+});
+
+export const friendsState = atom({
+  key: "friends",
+  default: friendsData.users,
+});
+
```
토글기능을 구현을 계획하고 user과 friends로 분리하신건가욤? 저도 나눌지 말지 고민했었는데 나누면 더 편한가요?

kwakseongjae replied:
입장시에 유저정보를 안받으면 조금 귀찮아지더라구요 
예를들어서 사용자가 유저목록 0번째 인덱스에 위치한다고 정하면 
나중에 이 앱이 범용적으로 사용될때 예상치 못한 결과가 나올거같아서 우선 뺴놨습니다 ! 
나중에 가능하면 입장시 유저정보를 받으면 해당 정보로 userData.json을 초기화하는 방식을 사용해볼까 합니다 !

물론 편한것도 있어요 ㅎㅎ

kimnabong2 commented on src/styles/App.styled.js:
```diff
@@ -0,0 +1,19 @@
+import styled from "styled-components";
+
+export const AppContainer = styled.div`
+  width: 512px;
+  height: 798px;
+
+  display: flex;
+  flex-direction: column;
+  margin: 0 auto; /* 페이지 중앙에 나타나도록 설정 */
+  margin-top: 50px;
+`;
+
+export const Container = styled.div`
+  width: 512px;
+  height: 768px;
+  background-color: #f4f4f4;
+
+  display: flex;
+`;
```
스타일 깔끔하게 빼내셨네요! 가독성 좋아요.

kwakseongjae committed:
REFACTOR: pages폴더에 각 페이지 컴포넌트 이동
-기존의 components에 몰아서 저장하던 방식에서 pages에 페이지 컴포넌트를 보관하는 방식으로 수정
-component폴더에는 각 페이지별로 폴더링해서 파일관리
-페이지 컴포넌트 이동으로 인한 경로 변경사항 수정

kwakseongjae committed:
FIX: 검색 기능 사용중 Enter키 입력시 다른 채팅이 다 날라가고 해당 유저의 채팅만 남는 버그 수정
-로컬스토리지를 이용하여 데이터를 관리하는 방식을 사용중 Enter키를 입력할 경우 해당 유저의 채팅 데이터를 제외한 나머지 데이터가 전부 삭제되는 버그 발견
-당장에는 Enter키 입력 자체가 불필요하다고 판단하여 components>ChattingList>ChattingListTopBar.js 파일에 엔터키 입력을 무시하는 코드 추가
